### PR TITLE
Update selenium section to use exposed host ports rather than dockerized service

### PR DESCRIPTION
**Pull request by rnorth**

### CODE REVIEW

File: step-9-selenium-tests.md [hunk 1]
```diff
@@ -1,91 +1,87 @@
...
+    }
 }
 ```
 
+## Component structure
+
+With the addition of this test code on top of `AbstractIntegrationTest`,we now have the following arrangement of components in our test:
+
+
+![Component structure diagram](https://g.gravizo.com/svg?
```
Comment (rnorth) [inline step-9-selenium-tests.md]: Need to fix diagram embedding - not working at present.

File: step-9-selenium-tests.md [hunk 2]
```diff
@@ -1,91 +1,91 @@
...
 ```
 
+## Component structure
+
+With the addition of this test code on top of `AbstractIntegrationTest`,we now have the following arrangement of components in our test:
+
+
+![Component structure diagram](https://kroki.io/graphviz/svg/eNp9UU1rAkEMvfdXhL1WD_VaVBBBUISitheRMjvGdXA6kSTbHor_vbt1dUddzW1e8j6SWbuMzX4LI_h9gqIkT4-A9bko8udL1SjLmxR9Nxl_TJPXM5jM9-xCBgMihTnyt7MYt8fvwSkoikoFH5qdOrdOi4JlKahxAVlgSHaHDDUS-byRaMYoNTIxm52pnzNcO4ljo8fg8q_nAdOPxLIXMRvXg3bvyvDuWJTi7kwU7eJcZe9RTlhWdyohJp-sTkYPOP-aTUGWljxxN_U5tk4f0D9L3uZq1DjyRA2rFNTDH-PTt5w=)
+<!--
+Regenerate if necessary at https://kroki.io/
```
Comment (bsideup) [inline step-9-selenium-tests.md]: 👍

File: step-9-selenium-tests.md [hunk 3]
```diff
@@ -1,91 +1,91 @@
...
+    @After
+    public void tearDown() {
+        browser.stop();
+    }
 }
 ```
 
+## Component structure
+
+With the addition of this test code on top of `AbstractIntegrationTest`,we now have the following arrangement of components in our test:
```
Comment (bsideup) [inline step-9-selenium-tests.md]:
```suggestion
With the addition of this test code on top of `AbstractIntegrationTest`, we now have the following arrangement of components in our test:
```

File: step-9-selenium-tests.md [hunk 4]
```diff
@@ -1,91 +1,91 @@
...
+
+* how to expose a host port to containers
+* once a host port is exposed, what URL should we tell the browser to access? (hint: it's not `localhost`)
+
+With this information, update the TODOs in the test class and see what happens.
+
+## Extra credit
+
+* The `BrowserWebDriverContainer` has a nice feature that allows test runs to be saved to a video, either for all tests or just failing tests. Check the documentation, and give this a try.
+* `BrowserWebDriverContainer` exposes a `getVncAddress()` method. Can you use this on your local machine to watch tests as they run via VNC? Even better, try and use the browser's inspect tools while stopped on a breakpoint in your tests!
```
Comment (bsideup) [inline step-9-selenium-tests.md]: I wonder if it makes sense to mention the approach from [my article](https://bsideup.github.io/posts/debugging_containers/) to run it on the same port

Comment (kiview) [reply step-9-selenium-tests.md]: I think it's a bit to much for the workshop, but we could just add a link for reference.

Comment (rnorth) [reply step-9-selenium-tests.md]: Agreed with both, let's add a link 😄

Comment (kiview) [reply step-9-selenium-tests.md]: Added

**Comment by kiview** [comment]
Argh, where does the conflict come from?

File: step-9-selenium-tests.md [hunk 5]
```diff
@@ -1,91 +1,92 @@
...
+* You can find some additional ideas [here](https://bsideup.github.io/posts/debugging_containers/) how to use a fixed port for local development and debugging by leveraging a tcp-proxy
```
Comment (rnorth) [inline step-9-selenium-tests.md]: Tiny tweaks:
```suggestion
* You can find some additional ideas [here](https://bsideup.github.io/posts/debugging_containers/) on how to use a fixed port for local development and debugging, leveraging a tcp-proxy.
```

**Pull request merged**